src/autoconf_pmt.c

```diff
@@ -553,7 +553,7 @@ int pmt_find_descriptor(uint8_t tag, unsigned char *buf, int descriptors_loop_le
 	while (descriptors_loop_len > 0)
 	{
 		unsigned char descriptor_tag = buf[0];
-		unsigned char descriptor_len = buf[1] + 2;
```

Comment:
I thing that `descriptor_len` variable name is misleading and the source of the problem.
According to specification, descriptor length is a byte value and is accessible via `buf[1]` whereas `buf[1] + 2` is the position of the next descriptor's tag and therefore can be larger that byte (0-255).

Reply:
I agree there is a potential confusion here 

here descriptor len is the total len including the header (1 byte for tag 1 byte for len) so maybe it could be called total descriptor len so something like this.